### PR TITLE
 Feature/Task-15: Implementend Supplier lookup 

### PR DESCRIPTION
**Pull request by 2musime**
# PR : Supplier lookup
## Description
This pull request adds the functionality for supplier search.
## Changes:

- [x] Search suppliers by SupplierName.
- [x] Search suppliers by SupplierIndex.
- [x] Search suppliers by City.
- [x] Retrieve the entire list of suppliers.

### CODE REVIEW

File: backend/CertificateManagerAPIs/Controllers/SupplierController.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public class SupplierController : Controller
+    {
+        private readonly ISupplierService _supplierService;
+
+        public SupplierController(ISupplierService supplierService)
+        {
+            _supplierService = supplierService;
+        }
+
+        [HttpGet("All suppliers")]
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/Controllers/SupplierController.cs]: method name must not contain empty spaces

**Review by elhasic — CHANGES_REQUESTED** [review]

File: backend/CertificateManagerAPIs/Controllers/SupplierController.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    [ApiController]
+    [Route("api/[controller]")]
+    public class SupplierController : Controller
+    {
+        private readonly ISupplierService _supplierService;
+
+        public SupplierController(ISupplierService supplierService)
+        {
+            _supplierService = supplierService;
+        }
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/Controllers/SupplierController.cs:17]: add new line

File: backend/CertificateManagerAPIs/Controllers/SupplierController.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        [HttpGet("All suppliers")]
+        public async Task<ActionResult<IEnumerable<SupplierDto>>> GetSuppliersAsync()
+        {
+            var suppliers = await _supplierService.GetSuppliersAsync();
+            return Ok(suppliers);
+        }
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/Controllers/SupplierController.cs:24]: add new line between each methods

File: backend/CertificateManagerAPIs/Controllers/SupplierController.cs (hunk 1 above)
Comment (elhasic) [reply backend/CertificateManagerAPIs/Controllers/SupplierController.cs]: this is not solved

**Review by elhasic — CHANGES_REQUESTED** [review]

File: backend/CertificateManagerAPIs/Controllers/SupplierController.cs (hunk 1 above)
Comment (2musime) [reply backend/CertificateManagerAPIs/Controllers/SupplierController.cs]: Thank you, I thought that it was the space above it but now I got it.

**Review by elhasic — APPROVED** [review]

**Pull request merged**